cstar/roms/component.py

```diff
@@ -920,202 +926,111 @@ def run(
-                    stdout=subprocess.PIPE,
-                    stderr=subprocess.PIPE,
-                    text=True,
-                )
+        else:  # cstar_sysmgr.scheduler is None
+            import time
 
-                # Process stdout line-by-line
-                tstep0 = 0
-                roms_init_string = ""
-                if romsprocess.stdout is None:
-                    raise RuntimeError("ROMS is not producing stdout")
-
-                # 2024-09-21 : the following is included as in some instances ROMS
-                # will exit with code 0 even if a fatal error occurs, see:
-                # https://github.com/CESR-lab/ucla-roms/issues/42
-
-                debugging = False  # Print raw output if true
-                if debugging:
-                    while True:
-                        output = romsprocess.stdout.readline()
-                        if output == "" and romsprocess.poll() is not None:
-                            break
-                        if output:
-                            print(output.strip())
-                else:
-                    for line in romsprocess.stdout:
-                        # Split the line by whitespace
-                        parts = line.split()
-
-                        # Check if there are exactly 9 parts and the first one is an integer
-                        if len(parts) == 9:
-                            try:
-                                # Try to convert the first part to an integer
-                                tstep = int(parts[0])
-                                if tstep0 == 0:
-                                    tstep0 = tstep
-                                    T0 = time.time()
-                                    # Capture the first integer and print it
-                                else:
-                                    ETA = (n_time_steps - (tstep - tstep0)) * (
-                                        (time.time() - T0) / (tstep - tstep0)
-                                    )
-                                    total_print_statements = 50
-                                    print_frq = (
-                                        n_time_steps // total_print_statements + 1
+            romsprocess = subprocess.Popen(
+                roms_exec_cmd,
+                shell=True,
+                cwd=run_path,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                text=True,
+            )
+
+            # Process stdout line-by-line
+            tstep0 = 0
+            roms_init_string = ""
+            if romsprocess.stdout is None:
+                raise RuntimeError("ROMS is not producing stdout")
+
+            # 2024-09-21 : the following is included as in some instances ROMS
+            # will exit with code 0 even if a fatal error occurs, see:
+            # https://github.com/CESR-lab/ucla-roms/issues/42
+
+            debugging = False  # Print raw output if true
```

Comment:
This is another example of cleaning that is needed (and is covered by 88) but is outside the scope of this PR (as `cstar_sysmgr.scheduler` is `None` for this block to be reached). There will be quite a few of these to get around to, this PR being the first and biggest example.

Reply:
Revisiting this code this isn't actually covered by issue 88: the "debugging" flag here is used to prevent C-Star parsing ROMS stdout in case something has gone wrong with ROMS.
This can be handled more gracefully, though - #203 should cover it.